### ISSUE TITLE
Wait For And Click Element with Image locator seems to locate the correct place, but clicks are not done with LMB

### ISSUE THREAD

**Issue by keljaus** [Accruent/robotframework-zoomba#175]
I installed openCV4NodeJS on top of Zoomba robot-framework library, so I could get the image locator into use.

However, even the image gets located, and "clicks" are done to the correct place of the application (close button), they seem to have no effect as they seem to not come from LMB, instead some kind of touch emulation?

Basically what happens, is that the test "clicks on air", never hitting the located spot

**Comment by Wolfe1** [comment]
Hey @keljaus,

So using the 'Click' type keywords should be using the mouse click to do the action, the touch actions would come through [Tap](https://accruent.github.io/robotframework-zoomba/DesktopLibraryDocumentation.html#Tap) or [Wait for and Tap](https://accruent.github.io/robotframework-zoomba/DesktopLibraryDocumentation.html#Wait%20For%20And%20Tap).

Anyway if you try a few things here maybe we can find a solution or understand the issue better:

1. Before your 'Wait For And Click Element' try sleeping for a few seconds before
2. After you first 'Wait For And Click Element' try a 'Click Element' on the same locator to see if it clicks
3. Try using [Wait For And Mouse Over And Click Element](https://accruent.github.io/robotframework-zoomba/DesktopLibraryDocumentation.html#Wait%20For%20And%20Mouse%20Over%20And%20Click%20Element) instead as it uses different actions to get to a similar result.
4. Try using [Wait For And Tap](https://accruent.github.io/robotframework-zoomba/DesktopLibraryDocumentation.html#Wait%20For%20And%20Tap) or [Wait For And Double Tap](https://accruent.github.io/robotframework-zoomba/DesktopLibraryDocumentation.html#Wait%20For%20And%20Double%20Tap)

Again hopefully one of those works or gives us a better idea on what is going on here.

**Comment by keljaus** [comment]
Cases 1 & 2 did the same as I described above.

---------------------------------------------------------

The case 3, returned: "FileNotFoundError: [Errno 2] No such file or directory: 'locator.png;'" and failed


**Here is the test of case 3:**

```
The case 3 with sleeps
	Sleep							  5
	Wait For And Mouse Over And Click Element		image=locator.png;
```

---------------------------------------------------------

The case 4, returned: "WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: An unknown error occurred in the remote end while processing the command." and failed

**Here is the test of case 4:**

```
The cases 4 both with Sleeps
	Sleep							  5
	Wait For And Tap				  image=locator.png
	Sleep							  5
	Wait For And Double Tap			  image=locator.png
```

-------------------------------------------------------

**Comment by keljaus** [comment]
Interesting was to notice that when I used the example calculator windows application instead of my custom WPF application, then the clicks were working correctly. However, I am not fully sure, if this because of the fact that the Calculator app supports the "tap inputs" I suspect the keyword is actually sending...or is there some problem with my WPF application's "close buttons".

My applications "close buttons" did seem to work when using different locator though:

```
Wait For And Click Element By Id Keyword Test
    Wait For And Click Element       accessibility_id=closeButton
```

^^ when I used this locator, it did the click on the button and it was effective. However, I would really need image recognition to work, for custom UI components and such, that may need workaround from image recognition.

**Comment by Wolfe1** [comment]
@keljaus 

In test 3 it seems you have an extra **";"** at the end of your locator, could you remove that and run your test again?

And for test 4 could you give me the log output from Appium when the error occurs?

-----------------------------------------------------------------

Does image recognition work for any other elements if you try? As you said it may just be the close button...perhaps you could use [Quit Application](https://accruent.github.io/robotframework-zoomba/DesktopLibraryDocumentation.html#Quit%20Application) if that window is a sub-window of the main app?

**Comment by keljaus** [comment]
@Wolfe1 The ";" at the end did not have any impact --> same error in case 3

The log output in case 4, is as such:

```
[Appium] Welcome to Appium v1.20.2
[Appium] Appium REST http interface listener started on 0.0.0.0:4723
[debug] [HTTP] Request idempotency key: a46a94ee-8348-4775-b650-99d8b2b42170
[HTTP] --> POST /wd/hub/session
[HTTP] {"capabilities":{"firstMatch":[{"appium:app":"Root","platformName":"Windows","appium:deviceName":"Windows","appium:newCommandTimeout":3600}]},"desiredCapabilities":{"app":"Root","platformName":"Windows","deviceName":"Windows","newCommandTimeout":3600}}
[debug] [W3C] Calling AppiumDriver.createSession() with args: [{"app":"Root","platformName":"Windows","deviceName":"Windows","newCommandTimeout":3600},null,{"firstMatch":[{"appium:app":"Root","platformName":"Windows","appium:deviceName":"Windows","appium:newCommandTimeout":3600}]}]
[debug] [BaseDriver] Event 'newSessionRequested' logged at 1612545344899 (19:15:44 GMT+0200 (Eastern European Standard Time))
[Appium] Appium v1.20.2 creating new WindowsDriver (v1.17.0) session
[debug] [BaseDriver] W3C capabilities and MJSONWP desired capabilities were provided
[debug] [BaseDriver] Creating session with W3C capabilities: {
[debug] [BaseDriver]   "alwaysMatch": {
[debug] [BaseDriver]     "platformName": "Windows",
[debug] [BaseDriver]     "appium:app": "Root",
[debug] [BaseDriver]     "appium:deviceName": "Windows",
[debug] [BaseDriver]     "appium:newCommandTimeout": 3600
[debug] [BaseDriver]   },
[debug] [BaseDriver]   "firstMatch": [
[debug] [BaseDriver]     {}
[debug] [BaseDriver]   ]
[debug] [BaseDriver] }
[BaseDriver] Session created with session id: 27e672ee-0999-424d-89ba-1ed44a30caf1
[WinAppDriver] WinAppDriver exists, but the checksum did not match. Was it replaced manually?
[WinAppDriver] Spawning 'C:\Program Files (x86)\Windows Application Driver\WinAppDriver.exe' with args: ["4724/wd/hub"]
[debug] [WinAppDriver] Windows Application Driver listening for requests at: http://127.0.0.1:4724/wd/hub
[debug] [WD Proxy] Matched '/status' to command name 'getStatus'
[debug] [WD Proxy] Proxying [GET /status] to [GET http://127.0.0.1:4724/wd/hub/status] with no body
[debug] [WinAppDriver] Press ENTER to exit.
[debug] [WinAppDriver] ==========================================
[debug] [WinAppDriver] GET /wd/hub/status HTTP/1.1
[debug] [WinAppDriver] Accept: application/json, */*
[debug] [WinAppDriver] Connection: keep-alive
[debug] [WinAppDriver] Content-Type: application/json; charset=utf-8
[debug] [WinAppDriver] Host: 127.0.0.1:4724
[debug] [WinAppDriver] User-Agent: appium
[debug] [WD Proxy] Got response with status 200: {"build":{"revision":"2003","time":"Wed Aug 26 07:56:06 2020","version":"1.2.2009"},"os":{"arch":"amd64","name":"windows","version":"10.0.18363"}}
[debug] [WinAppDriver] Starting WinAppDriver session. Will timeout in '20000' ms.
[debug] [WD Proxy] Matched '/session' to command name 'createSession'
[debug] [WD Proxy] Proxying [POST /session] to [POST http://127.0.0.1:4724/wd/hub/session] with body: {"desiredCapabilities":{"platformName":"Windows","app":"Root","deviceName":"Windows","newCommandTimeout":3600}}
[debug] [WinAppDriver] HTTP/1.1 200 OK
[debug] [WinAppDriver] Content-Length: 146
[debug] [WinAppDriver] Content-Type: application/json
[debug] [WinAppDriver]
[debug] [WinAppDriver] {"build":{"revision":"2003","time":"Wed Aug 26 07:56:06 2020","version":"1.2.2009"},"os":{"arch":"amd64","name":"windows","version":"10.0.18363"}}
[debug] [WinAppDriver] ==========================================
[debug] [WinAppDriver] POST /wd/hub/session HTTP/1.1
[debug] [WinAppDriver] Accept: application/json, */*
[debug] [WinAppDriver] Connection: keep-alive
[debug] [WinAppDriver] Content-Length: 111
[debug] [WinAppDriver] Content-Type: application/json; charset=utf-8
[debug] [WinAppDriver] Host: 127.0.0.1:4724
[debug] [WinAppDriver] User-Agent: appium
[debug] [WinAppDriver]
[debug] [WinAppDriver] {"desiredCapabilities":{"platformName":"Windows","app":"Root","deviceName":"Windows","newCommandTimeout":3600}}
[debug] [WinAppDriver] HTTP/1.1 200 OK
[debug] [WinAppDriver] Content-Length: 111
[debug] [WinAppDriver] Content-Type: application/json
[debug] [WinAppDriver]
[debug] [WinAppDriver] {"sessionId":"61FD9FC2-4D1C-411D-A726-8750E1CCBCE9","status":0,"value":{"app":"Root","platformName":"Windows"}}
[debug] [WD Proxy] Got response with status 200: {"sessionId":"61FD9FC2-4D1C-411D-A726-8750E1CCBCE9","status":0,"value":{"app":"Root","platformName":"Windows"}}
[WD Proxy] Determined the downstream protocol as 'MJSONWP'
[Appium] New WindowsDriver session created successfully, session 27e672ee-0999-424d-89ba-1ed44a30caf1 added to master session list
[debug] [BaseDriver] Event 'newSessionStarted' logged at 1612545345122 (19:15:45 GMT+0200 (Eastern European Standard Time))
[debug] [W3C (27e672ee)] Cached the protocol value 'W3C' for the new session 27e672ee-0999-424d-89ba-1ed44a30caf1
[debug] [W3C (27e672ee)] Responding to client with driver.createSession() result: {"capabilities":{"platformName":"Windows","app":"Root","deviceName":"Windows","newCommandTimeout":3600}}
[HTTP] <-- POST /wd/hub/session 200 228 ms - 165
[HTTP]
[debug] [HTTP] Request idempotency key: 341788d1-fc98-41ca-9b53-29500a966e7a
[HTTP] --> POST /wd/hub/session
[HTTP] {"capabilities":{"firstMatch":[{"platformName":"Windows","appium:deviceName":"Windows","appium:app":"xxxxxxxxxxxxxxxxxxxxxxxx"}]},"desiredCapabilities":{"platformName":"Windows","deviceName":"Windows","app":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"}}
[debug] [W3C] Calling AppiumDriver.createSession() with args: [{"platformName":"Windows","deviceName":"Windows","app":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"},null,{"firstMatch":[{"platformName":"Windows","appium:deviceName":"Windows","appium:app":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"}]}]
[debug] [BaseDriver] Event 'newSessionRequested' logged at 1612545345150 (19:15:45 GMT+0200 (Eastern European Standard Time))
[Appium] Appium v1.20.2 creating new WindowsDriver (v1.17.0) session
[debug] [BaseDriver] W3C capabilities and MJSONWP desired capabilities were provided
[debug] [BaseDriver] Creating session with W3C capabilities: {
[debug] [BaseDriver]   "alwaysMatch": {
[debug] [BaseDriver]     "platformName": "Windows",
[debug] [BaseDriver]     "appium:deviceName": "Windows",
[debug] [BaseDriver]     "appium:app": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
[debug] [BaseDriver]   },
[debug] [BaseDriver]   "firstMatch": [
[debug] [BaseDriver]     {}
[debug] [BaseDriver]   ]
[debug] [BaseDriver] }
[BaseDriver] Session created with session id: e3268737-5c4c-4554-8864-8ab06b36d647
[WinAppDriver] WinAppDriver exists, but the checksum did not match. Was it replaced manually?
[WinAppDriver] Spawning 'C:\Program Files (x86)\Windows Application Driver\WinAppDriver.exe' with args: ["4725/wd/hub"]
[debug] [WD Proxy] Matched '/status' to command name 'getStatus'
[debug] [WD Proxy] Proxying [GET /status] to [GET http://127.0.0.1:4725/wd/hub/status] with no body
[WD Proxy] connect ECONNREFUSED 127.0.0.1:4725
[debug] [WinAppDriver] Windows Application Driver listening for requests at: http://127.0.0.1:4725/wd/hub
[debug] [WinAppDriver] Press ENTER to exit.
[debug] [WD Proxy] Matched '/status' to command name 'getStatus'
[debug] [WD Proxy] Proxying [GET /status] to [GET http://127.0.0.1:4725/wd/hub/status] with no body
[debug] [WinAppDriver] ==========================================
[debug] [WinAppDriver] GET /wd/hub/status HTTP/1.1
[debug] [WinAppDriver] Accept: application/json, */*
[debug] [WinAppDriver] Connection: keep-alive
[debug] [WinAppDriver] Content-Type: application/json; charset=utf-8
[debug] [WinAppDriver] Host: 127.0.0.1:4725
[debug] [WinAppDriver] User-Agent: appium
[debug] [WD Proxy] Got response with status 200: {"build":{"revision":"2003","time":"Wed Aug 26 07:56:06 2020","version":"1.2.2009"},"os":{"arch":"amd64","name":"windows","version":"10.0.18363"}}
[debug] [WinAppDriver] Starting WinAppDriver session. Will timeout in '20000' ms.
[debug] [WD Proxy] Matched '/session' to command name 'createSession'
[debug] [WD Proxy] Proxying [POST /session] to [POST http://127.0.0.1:4725/wd/hub/session] with body: {"desiredCapabilities":{"platformName":"Windows","deviceName":"Windows","app":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"}}
[debug] [WinAppDriver] HTTP/1.1 200 OK
[debug] [WinAppDriver] Content-Length: 146
[debug] [WinAppDriver] Content-Type: application/json
[debug] [WinAppDriver]
[debug] [WinAppDriver] {"build":{"revision":"2003","time":"Wed Aug 26 07:56:06 2020","version":"1.2.2009"},"os":{"arch":"amd64","name":"windows","version":"10.0.18363"}}
[debug] [WinAppDriver] ==========================================
[debug] [WinAppDriver] POST /wd/hub/session HTTP/1.1
[debug] [WinAppDriver] Accept: application/json, */*
[debug] [WinAppDriver] Connection: keep-alive
[debug] [WinAppDriver] Content-Length: 159
[debug] [WinAppDriver] Content-Type: application/json; charset=utf-8
[debug] [WinAppDriver] Host: 127.0.0.1:4725
[debug] [WinAppDriver] User-Agent: appium
[debug] [WinAppDriver]
[debug] [WinAppDriver] {"desiredCapabilities":{"platformName":"Windows","deviceName":"Windows","app":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxe"}}
[debug] [WinAppDriver] HTTP/1.1 200 OK
[debug] [WinAppDriver] Content-Length: 184
[debug] [WinAppDriver] Content-Type: application/json
[debug] [WinAppDriver]
[debug] [WinAppDriver] {"sessionId":"B390765D-C8BF-426F-8EC0-83535B9F2AAA","status":0,"value":{"app":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"}}
[debug] [WD Proxy] Got response with status 200: {"sessionId":"B390765D-C8BF-426F-8EC0-83535B9F2AAA","status":0,"value":{"app":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"}}
[WD Proxy] Determined the downstream protocol as 'MJSONWP'
[Appium] New WindowsDriver session created successfully, session e3268737-5c4c-4554-8864-8ab06b36d647 added to master session list
[debug] [BaseDriver] Event 'newSessionStarted' logged at 1612545349309 (19:15:49 GMT+0200 (Eastern European Standard Time))
[debug] [W3C (e3268737)] Cached the protocol value 'W3C' for the new session e3268737-5c4c-4554-8864-8ab06b36d647
[debug] [W3C (e3268737)] Responding to client with driver.createSession() result: {"capabilities":{"platformName":"Windows","deviceName":"Windows","app":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"}}
[HTTP] <-- POST /wd/hub/session 200 4162 ms - 213
[HTTP]
[HTTP] --> POST /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/window/maximize
[HTTP] {}
[W3C (e3268737)] Driver proxy active, passing request on via HTTP proxy
[debug] [WD Proxy] Matched '/wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/window/maximize' to command name 'maximizeWindow'
[debug] [WD Proxy] Proxying [POST /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/window/maximize] to [POST http://127.0.0.1:4725/wd/hub/session/B390765D-C8BF-426F-8EC0-83535B9F2AAA/window/maximize] with body: {}
[debug] [WinAppDriver] ==========================================
[debug] [WinAppDriver] POST /wd/hub/session/B390765D-C8BF-426F-8EC0-83535B9F2AAA/window/maximize HTTP/1.1
[debug] [WinAppDriver] Accept: application/json, */*
[debug] [WinAppDriver] Connection: keep-alive
[debug] [WinAppDriver] Content-Length: 2
[debug] [WinAppDriver] Content-Type: application/json; charset=utf-8
[debug] [WinAppDriver] Host: 127.0.0.1:4725
[debug] [WinAppDriver] User-Agent: appium
[debug] [WinAppDriver]
[debug] [WinAppDriver] {}
[debug] [WinAppDriver] HTTP/1.1 200 OK
[debug] [WinAppDriver] Content-Length: 63
[debug] [WinAppDriver] Content-Type: application/json
[debug] [WinAppDriver]
[debug] [WinAppDriver] {"sessionId":"B390765D-C8BF-426F-8EC0-83535B9F2AAA","status":0}
[debug] [WD Proxy] Got response with status 200: {"sessionId":"B390765D-C8BF-426F-8EC0-83535B9F2AAA","status":0}
[WD Proxy] Replacing sessionId B390765D-C8BF-426F-8EC0-83535B9F2AAA with e3268737-5c4c-4554-8864-8ab06b36d647
[HTTP] <-- POST /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/window/maximize 200 90 ms - 65
[HTTP]
[HTTP] --> POST /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/appium/app/close
[HTTP] {}
[W3C (e3268737)] Driver proxy active, passing request on via HTTP proxy
[debug] [WD Proxy] Matched '/wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/appium/app/close' to command name 'closeApp'
[debug] [WD Proxy] Proxying [POST /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/appium/app/close] to [POST http://127.0.0.1:4725/wd/hub/session/B390765D-C8BF-426F-8EC0-83535B9F2AAA/appium/app/close] with body: {}
[debug] [WinAppDriver] ==========================================
[debug] [WinAppDriver] POST /wd/hub/session/B390765D-C8BF-426F-8EC0-83535B9F2AAA/appium/app/close HTTP/1.1
[debug] [WinAppDriver] Accept: application/json, */*
[debug] [WinAppDriver] Connection: keep-alive
[debug] [WinAppDriver] Content-Length: 2
[debug] [WinAppDriver] Content-Type: application/json; charset=utf-8
[debug] [WinAppDriver] Host: 127.0.0.1:4725
[debug] [WinAppDriver] User-Agent: appium
[debug] [WinAppDriver]
[debug] [WinAppDriver] {}
[debug] [WinAppDriver] HTTP/1.1 200 OK
[debug] [WinAppDriver] Content-Length: 63
[debug] [WinAppDriver] Content-Type: application/json
[debug] [WinAppDriver]
[debug] [WinAppDriver] {"sessionId":"B390765D-C8BF-426F-8EC0-83535B9F2AAA","status":0}
[debug] [WD Proxy] Got response with status 200: {"sessionId":"B390765D-C8BF-426F-8EC0-83535B9F2AAA","status":0}
[WD Proxy] Replacing sessionId B390765D-C8BF-426F-8EC0-83535B9F2AAA with e3268737-5c4c-4554-8864-8ab06b36d647
[HTTP] <-- POST /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/appium/app/close 200 48 ms - 65
[HTTP]
[HTTP] --> POST /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/appium/app/launch
[HTTP] {}
[W3C (e3268737)] Driver proxy active, passing request on via HTTP proxy
[debug] [WD Proxy] Matched '/wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/appium/app/launch' to command name 'launchApp'
[debug] [WD Proxy] Proxying [POST /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/appium/app/launch] to [POST http://127.0.0.1:4725/wd/hub/session/B390765D-C8BF-426F-8EC0-83535B9F2AAA/appium/app/launch] with body: {}
[debug] [WinAppDriver] ==========================================
[debug] [WinAppDriver] POST /wd/hub/session/B390765D-C8BF-426F-8EC0-83535B9F2AAA/appium/app/launch HTTP/1.1
[debug] [WinAppDriver] Accept: application/json, */*
[debug] [WinAppDriver] Connection: keep-alive
[debug] [WinAppDriver] Content-Length: 2
[debug] [WinAppDriver] Content-Type: application/json; charset=utf-8
[debug] [WinAppDriver] Host: 127.0.0.1:4725
[debug] [WinAppDriver] User-Agent: appium
[debug] [WinAppDriver]
[debug] [WinAppDriver] {}
[debug] [WinAppDriver] HTTP/1.1 200 OK
[debug] [WinAppDriver] Content-Length: 63
[debug] [WinAppDriver] Content-Type: application/json
[debug] [WinAppDriver]
[debug] [WinAppDriver] {"sessionId":"B390765D-C8BF-426F-8EC0-83535B9F2AAA","status":0}
[debug] [WD Proxy] Got response with status 200: {"sessionId":"B390765D-C8BF-426F-8EC0-83535B9F2AAA","status":0}
[WD Proxy] Replacing sessionId B390765D-C8BF-426F-8EC0-83535B9F2AAA with e3268737-5c4c-4554-8864-8ab06b36d647
[HTTP] <-- POST /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/appium/app/launch 200 2823 ms - 65
[HTTP]
[HTTP] --> GET /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/screenshot
[HTTP] {}
[W3C (e3268737)] Driver proxy active, passing request on via HTTP proxy
[debug] [WD Proxy] Matched '/wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/screenshot' to command name 'getScreenshot'
[debug] [WD Proxy] Proxying [GET /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/screenshot] to [GET http://127.0.0.1:4725/wd/hub/session/B390765D-C8BF-426F-8EC0-83535B9F2AAA/screenshot] with no body
[debug] [WinAppDriver] ==========================================
[debug] [WinAppDriver] GET /wd/hub/session/B390765D-C8BF-426F-8EC0-83535B9F2AAA/screenshot HTTP/1.1
[debug] [WinAppDriver] Accept: application/json, */*
[debug] [WinAppDriver] Connection: keep-alive
[debug] [WinAppDriver] Content-Type: application/json; charset=utf-8
[debug] [WinAppDriver] Host: 127.0.0.1:4725
[debug] [WinAppDriver] User-Agent: appium
[debug] [WD Proxy] Got response with status 200: {"sessionId":"B390765D-C8BF-426F-8EC0-83535B9F2AAA","status":0,"value":"iVBORw0KGgoAAAANSUhEUgAAB4AAAAQQCAIAAACMTtFPAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAP+6SURBVHhe7N19nFXVned7SiQS2kQT01IKVNk3mvGlEylSRKGI/fL2g4KQiPfy0KZnkoEuCFEYp5POvS/6EkTClVe/EvMaXzCKUA2d7pu2ebg3miAPZqaT7rRo0iCFYztOa9KhgKYKtSM+EFQi98v+rVq92E9nn1PnVNUpPu8/Nmutvfbaa6/9cGov1llnGAAAAAAAAAAAtdDg/i1HS0vLxRdfrMCpU6f+8R//8dixY5YOAAAAAAAAAIA33P1bWENDwx/8wR989KMfveKKK37jN37jjTfe6OrqcusAAAAAAAAAAOh1nvs3V0PEwueff/6IESMsrMT3ve99ftV55xUqDQAAAAAAAABwLig9AnrEiBETJkwYO3bs66+/Pnr06La2tjFjxrh1w4ZdcMEFDQ0Nv/jFL5ShtbX15MmTb7zxhlsHAAAAAAAAADiHlZ4Durm5+bOf/ezw4cNff/31ESNGXHjhhX7Iszl16tQvfvGLUZGurq6//Mu/fPvtt906AAAAAAAAAMC5qsQI6PPPP/9//9//9w9/+MPnnXfe+9//fhvv7Nb10qpf+7Vfs7k4Ro0adfz48Z6eHrcOAAAAAAAAAHCuKtEBfcUVV9x4440u0uvNN9/8h3/4h4MHD7799tsXXXRROPXz8OHDT5069dOf/vRXv/qVSwIAAAAAAAAAnJNKdED/8pe/vOCCC0aPHu17mU+cOPEXf/EXzz777EuR888///LLL/fDon/xi1/83d/93csvv2zRKvr+97/f09Pz05/+1MVRPrXh5z73ufe///379u1zSVWyadMmXSrJs1P8rGWVMICsuT72sY/94Ac/cEkAAAAAAADAOWnTpk2LFy/+i7/4C...
[WD Proxy] Replacing sessionId B390765D-C8BF-426F-8EC0-83535B9F2AAA with e3268737-5c4c-4554-8864-8ab06b36d647
[HTTP] <-- GET /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/screenshot 200 62 ms - 167263
[HTTP]
[debug] [WinAppDriver] HTTP/1.1 200 OK
[debug] [WinAppDriver] Content-Length: 167274
[debug] [WinAppDriver] Content-Type: application/json
[HTTP] --> POST /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/appium/app/close
[HTTP] {}
[W3C (e3268737)] Driver proxy active, passing request on via HTTP proxy
[debug] [WD Proxy] Matched '/wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/appium/app/close' to command name 'closeApp'
[debug] [WD Proxy] Proxying [POST /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/appium/app/close] to [POST http://127.0.0.1:4725/wd/hub/session/B390765D-C8BF-426F-8EC0-83535B9F2AAA/appium/app/close] with body: {}
[debug] [WinAppDriver] ==========================================
[debug] [WinAppDriver] POST /wd/hub/session/B390765D-C8BF-426F-8EC0-83535B9F2AAA/appium/app/close HTTP/1.1
[debug] [WinAppDriver] Accept: application/json, */*
[debug] [WinAppDriver] Connection: keep-alive
[debug] [WinAppDriver] Content-Length: 2
[debug] [WinAppDriver] Content-Type: application/json; charset=utf-8
[debug] [WinAppDriver] Host: 127.0.0.1:4725
[debug] [WinAppDriver] User-Agent: appium
[debug] [WinAppDriver] HTTP/1.1 200 OK
[debug] [WinAppDriver] Content-Length: 63
[debug] [WinAppDriver] Content-Type: application/json
[debug] [WinAppDriver]
[debug] [WinAppDriver] {"sessionId":"B390765D-C8BF-426F-8EC0-83535B9F2AAA","status":0}
[debug] [WD Proxy] Got response with status 200: {"sessionId":"B390765D-C8BF-426F-8EC0-83535B9F2AAA","status":0}
[WD Proxy] Replacing sessionId B390765D-C8BF-426F-8EC0-83535B9F2AAA with e3268737-5c4c-4554-8864-8ab06b36d647
[HTTP] <-- POST /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647/appium/app/close 200 53 ms - 65
[HTTP]
[HTTP] --> DELETE /wd/hub/session/27e672ee-0999-424d-89ba-1ed44a30caf1
[HTTP] {}
[debug] [W3C (27e672ee)] Calling AppiumDriver.deleteSession() with args: ["27e672ee-0999-424d-89ba-1ed44a30caf1"]
[debug] [BaseDriver] Event 'quitSessionRequested' logged at 1612545357409 (19:15:57 GMT+0200 (Eastern European Standard Time))
[Appium] Removing session 27e672ee-0999-424d-89ba-1ed44a30caf1 from our master session list
[debug] [WinAppDriver] Deleting WinAppDriver session
[debug] [WinAppDriver] Deleting WinAppDriver server session
[debug] [WD Proxy] Matched '/session/61FD9FC2-4D1C-411D-A726-8750E1CCBCE9' to command name 'deleteSession'
[debug] [WD Proxy] Proxying [DELETE /session/61FD9FC2-4D1C-411D-A726-8750E1CCBCE9] to [DELETE http://127.0.0.1:4724/wd/hub/session/61FD9FC2-4D1C-411D-A726-8750E1CCBCE9] with no body
[debug] [WinAppDriver] ==========================================
[debug] [WinAppDriver] DELETE /wd/hub/session/61FD9FC2-4D1C-411D-A726-8750E1CCBCE9 HTTP/1.1
[debug] [WinAppDriver] Accept: application/json, */*
[debug] [WinAppDriver] Connection: keep-alive
[debug] [WinAppDriver] Content-Type: application/json; charset=utf-8
[debug] [WinAppDriver] Host: 127.0.0.1:4724
[debug] [WinAppDriver] User-Agent: appium
[debug] [WD Proxy] Got response with status 200: {"status":0}
[debug] [WinAppDriver] HTTP/1.1 200 OK
[debug] [WinAppDriver] Content-Length: 12
[debug] [WinAppDriver] Content-Type: application/json
[debug] [WinAppDriver]
[debug] [WinAppDriver] {"status":0}
[WinAppDriver] WinAppDriver exited with code null, signal SIGTERM
[debug] [BaseDriver] Event 'quitSessionFinished' logged at 1612545357431 (19:15:57 GMT+0200 (Eastern European Standard Time))
[debug] [W3C (27e672ee)] Received response: null
[debug] [W3C (27e672ee)] But deleting session, so not returning
[debug] [W3C (27e672ee)] Responding to client with driver.deleteSession() result: null
[HTTP] <-- DELETE /wd/hub/session/27e672ee-0999-424d-89ba-1ed44a30caf1 200 25 ms - 14
[HTTP]
[HTTP] --> DELETE /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647
[HTTP] {}
[debug] [W3C (e3268737)] Calling AppiumDriver.deleteSession() with args: ["e3268737-5c4c-4554-8864-8ab06b36d647"]
[debug] [BaseDriver] Event 'quitSessionRequested' logged at 1612545357434 (19:15:57 GMT+0200 (Eastern European Standard Time))
[Appium] Removing session e3268737-5c4c-4554-8864-8ab06b36d647 from our master session list
[debug] [WinAppDriver] Deleting WinAppDriver session
[debug] [WinAppDriver] Deleting WinAppDriver server session
[debug] [WD Proxy] Matched '/session/B390765D-C8BF-426F-8EC0-83535B9F2AAA' to command name 'deleteSession'
[debug] [WD Proxy] Proxying [DELETE /session/B390765D-C8BF-426F-8EC0-83535B9F2AAA] to [DELETE http://127.0.0.1:4725/wd/hub/session/B390765D-C8BF-426F-8EC0-83535B9F2AAA] with no body
[debug] [WinAppDriver] ==========================================
[debug] [WinAppDriver] DELETE /wd/hub/session/B390765D-C8BF-426F-8EC0-83535B9F2AAA HTTP/1.1
[debug] [WinAppDriver] Accept: application/json, */*
[debug] [WinAppDriver] Connection: keep-alive
[debug] [WinAppDriver] Content-Type: application/json; charset=utf-8
[debug] [WinAppDriver] Host: 127.0.0.1:4725
[debug] [WinAppDriver] User-Agent: appium
[debug] [WD Proxy] Got response with status 200: {"status":0}
[debug] [WinAppDriver] HTTP/1.1 200 OK
[debug] [WinAppDriver] Content-Length: 12
[debug] [WinAppDriver] Content-Type: application/json
[debug] [WinAppDriver]
[debug] [WinAppDriver] {"status":0}
[WinAppDriver] WinAppDriver exited with code null, signal SIGTERM
[debug] [BaseDriver] Event 'quitSessionFinished' logged at 1612545357459 (19:15:57 GMT+0200 (Eastern European Standard Time))
[debug] [W3C (e3268737)] Received response: null
[debug] [W3C (e3268737)] But deleting session, so not returning
[debug] [W3C (e3268737)] Responding to client with driver.deleteSession() result: null
[HTTP] <-- DELETE /wd/hub/session/e3268737-5c4c-4554-8864-8ab06b36d647 200 27 ms - 14
[HTTP]
```

**Comment by Wolfe1** [comment]
@keljaus So during some investigation on another ticket I learned some things that may be pertinent to this issue.

First of all it seemed that in Appium starting at version 1.20.0 the mouse type keywords stopped working due to a change in the way the commands were being interpreted. I have just released a change that will revert this behavior back for now but apparently this workaround will not work once Appium 2.0 is released in the future.

https://github.com/Accruent/robotframework-zoomba/issues/179

Second through looking into his it seems that WInAppDriver never supported clicks but instead everything was translated into touch and pen movements which may describe why you are seeing these touch type motions.

1. What version of appium are you using? I am guessing not latest since before the recent update to this library it would not have worked.
2. Perhaps try updating to latest zoomba and latest appium and try again to see if there is any change or improvement here.

Thanks!
